node/libs/roles/src/validator/messages/leader_prepare.rs

```diff
@@ -80,24 +100,33 @@ impl PrepareQC {
 
     /// Add a validator's signed message.
     /// Message is assumed to be already verified.
-    // TODO: check if there is already a message from that validator.
```

Comment:
oh, I now remember that this TODO is about: we just check here that we don't have a signature from this validator on msg, but since PrepareQC collects signatures for various messages, we should check that we don't have a signature from this validator on ANY msg in the certificate.

Reply:
+1